node/silkworm/downloader/internals/db_utils.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2021-2022 The Silkworm Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "db_utils.hpp"
+
+#include <silkworm/db/tables.hpp>
+#include <silkworm/db/util.hpp>
+
+namespace silkworm {
+
+// Read all headers up to limit, in reverse order from last, processing each via a user defined callback
+// alternative implementation: use cursor_for_count(cursor, const WalkFunc&, size_t max_count, CursorMoveDirection)
+void read_headers_in_reverse_order(mdbx::txn& txn, size_t limit, std::function<void(BlockHeader&&)> callback) {
+    auto header_table = db::open_cursor(txn, db::table::kHeaders);
```

Comment:
Should use db:Cursor ctor to take advantage of cursors pool

Reply:
ok